xcm/pallet-xcm/src/mock.rs

```diff
@@ -75,7 +75,7 @@ impl SendXcm for TestSendXcm {
 pub struct TestSendXcmErrX8;
 impl SendXcm for TestSendXcmErrX8 {
 	fn send_xcm(dest: MultiLocation, msg: Xcm) -> XcmResult {
-		if let MultiLocation::X8(..) = dest {
+		if dest.len() >= 8 {
```

Comment:
```suggestion
		if dest.len() == 8 {
```

I think this is actually the proper change